### PR TITLE
Add py-yt 4.x versions

### PR DESCRIPTION
**Pull request by charmoniumQ**
Add py-yt 4.x versions to the builtin Spack package repository.

This is relevant especially because of recent issues that only work in 4.x, e.g. yt-project/yt/#905.

### CODE REVIEW

**Comment by charmoniumQ** [comment]
```
$ ./bin/spack style
==> Running style checks on spack
  selected: isort, mypy, flake8
==> Modified files
==> Running isort checks
  isort checks were clean
==> Running mypy checks
  Success: no issues found in 556 source files
  mypy checks were clean
==> Running flake8 checks
  flake8 checks were clean
==> spack style checks were clean
```

I think this a spurious failure? How do I rerun that job?

**Comment by qobilidop** [comment]
Could you also update the following as they seem out of date too?
- `url      = "https://github.com/yt-project/yt/archive/yt-3.5.0.tar.gz"` -> `url      = "https://github.com/yt-project/yt/archive/yt-4.0.0.tar.gz"`
- `version("develop", branch="master")` -> `version("develop", branch="main")`
- Remove `version("develop-4.0", branch="yt-4.0")`

Other changes LGTM. Thanks!

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Dependencies need to be updated:
```toml
requires = [
  "setuptools>=19.6",
  # see https://github.com/numpy/numpy/pull/18389
  "wheel>=0.36.2",

  # cython version is imposed by that of numpy, see release notes
  # https://github.com/numpy/numpy/releases/tag/v1.19.2
  # Cython 3.0 is the next version after 0.29, and a major change,
  # we forbid it until we can properly test against it
  "Cython>=0.26.1,<3.0; python_version=='3.6'",
  "Cython>=0.29.21,<3.0; python_version>='3.7'",
  "oldest-supported-numpy",
]
```
```cfg
install_requires =
    cmyt>=0.2.2
    matplotlib!=3.4.2,>=2.1.0 # keep in sync with tests/windows_conda_requirements.txt
    more-itertools>=8.4
    numpy>=1.13.3
    packaging>=20.9
    setuptools>=19.6
    tomli>=1.2.3
    tomli-w>=0.4.0
    tqdm>=3.4.0
    unyt>=2.8.0
python_requires = >=3.6,<3.12
```
For the version think, can you add `deprecated=True` to `develop` and `develop-4.0` and add a new version `main` that points to `main`? That will be less confusing. URL isn't actually important since Spack automatically replaces the version number with the correct version, but you can change if you want. Let me know if you need help with any of this.

**Comment by charmoniumQ** [comment]
I noticed when I tried to use `yt`. I added unit tests, added those dependent packages, updated the `depends_on`.

@qobilidop, Is it ok that I put you down as a maintainer of `py-unyt` and `py-cmyt`? They are spinoffs of `yt` and quite related.

I'm not sure how much value there is in supporting 4.0.{0,1,2} given that 4.0.3 is supported and they have different dependencies.

**Comment by qobilidop** [comment]
> @qobilidop, Is it ok that I put you down as a maintainer of `py-unyt` and `py-cmyt`? They are spinoffs of `yt` and quite related.

That's fine to me.

File: var/spack/repos/builtin/packages/py-cmyt/package.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class PyCmyt(PythonPackage):
+    """Matplotlib colormaps from the yt project !"""
+    homepage = "https://yt-project.org"
+    url      = "https://github.com/yt-project/cmyt/archive/v1.0.4.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmyt/package.py]:
```suggestion
    pypi = "cmyt/cmyt-1.0.4.tar.gz"
```
We prefer to download from PyPI instead of GitHub for the reasons outlined [here](https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github). Note that this will likely change the checksums. You can run `spack checksum py-cmyt` to compute the new checksums then copy-n-paste them.

File: var/spack/repos/builtin/packages/py-cmyt/package.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    git      = "https://github.com/yt-project/cmyt.git"
+
+    maintainers = ['qobilidop', 'charmoniumq']
+
+    version("develop", branch="main")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmyt/package.py]:
```suggestion
    version("main", branch="main")
```

File: var/spack/repos/builtin/packages/py-cmyt/package.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    version("1.0.4", "c833ed24eadca8c788e59ce27119d26b2e7d4c6af83293e69385412629294b2a")
+    version("1.0.3", "7d071f86cd9e53ab4f2af9f1b6539d632f52c0d453caf93a80376105ac351556")
+    version("1.0.2", "4cf2e5f06031141e03ab9301a66c99be024a1a8c35ed99d1a88a24baf754234e")
+    version("1.0.1", "c20ec811c0cd71ffb3482e0d269bd22e0f088a3187dd615c59edf348cbefded1")
+    version("1.0.0", "64751ecd5916efe817beb92d0b5b34b0dc134c8ee0dab3f212cb089b6b78bbd4")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmyt/package.py]: Add this:
```python
depends_on('py-setuptools', type='build')
```

File: var/spack/repos/builtin/packages/py-cmyt/package.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    depends_on("py-colorspacious@1.1.2:")
+    depends_on("py-matplotlib@2.1.0:")
+    depends_on("py-more-itertools@8.4.0:")
+    depends_on("py-numpy@1.13.3:")
+    depends_on("py-typing-extensions@3.10.0.2:", when="^python@:3.8")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmyt/package.py]: Spack version ranges are inclusive, so this should actually be `^python@:3.7`. Also, all these deps should have `type=('build', 'run')`. See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#setuptools.

File: var/spack/repos/builtin/packages/py-tomli/package.py [hunk 5]
```diff
@@ -15,6 +15,10 @@ class PyTomli(PythonPackage):
     url = "https://files.pythonhosted.org/packages/py3/t/tomli/tomli-1.2.2-py3-none-any.whl"
     list_url = "https://pypi.org/simple/tomli/"
 
+    version('2.0.1', sha256='939de3e7a6161af0c887ef91b7d41a53e7c5a1ca976325f429cb46ea9bc30ecc', expand=False)
+    version('2.0.0', sha256='b5bde28da1fed24b9bd1d4d2b8cba62300bfb4ec9a6187a957e8ddb9434c5224', expand=False)
+    version('1.2.3', sha256='e3069e4be3ead9668e21cb9b074cd948f7b3113fd9c8bba083f48247aab8b11c', expand=False)
+    version('1.2.2', sha256='f04066f68f5554911363063a30b108d2b5a5b1a010aa8b6132af78489fe3aade', expand=False)
     version('1.2.2', sha256='f04066f68f5554911363063a30b108d2b5a5b1a010aa8b6132af78489fe3aade', expand=False)
     version('1.2.1', sha256='8dd0e9524d6f386271a36b41dbf6c57d8e32fd96fd22b6584679dc569d20899f', expand=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tomli/package.py:19]: If you unzip the wheel and look in `tomli-2.0.1.dist-info/METADATA` the Python dep is now 3.7+. Not sure which version this changed in. See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#wheels

File: var/spack/repos/builtin/packages/py-unyt/package.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+from spack import *
+
+
+class PyUnyt(PythonPackage):
+    """A package for handling numpy arrays with units.
+
+    """
+    homepage = "https://yt-project.org"
+    url      = "https://github.com/yt-project/unyt/archive/v2.8.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-unyt/package.py]:
```suggestion
   pypi = "unyt/unyt-2.8.0.tar.gz"
```
Update checksums too

File: var/spack/repos/builtin/packages/py-unyt/package.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    git      = "https://github.com/yt-project/unyt.git"
+
+    maintainers = ['qobilidop', 'charmoniumq']
+
+    version("develop", branch="main")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-unyt/package.py]:
```suggestion
    version("main", branch="main")
```

File: var/spack/repos/builtin/packages/py-unyt/package.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    version("2.8.0", sha256="4fc95704cba4527604f21724d0d9845aea07831a6a2a3d1be9b9d18395e0c345")
+    version("2.7.2", sha256="9ad36f890549d18ecb9926019e96c4d27d380e909895a89ac7f1fb168a451edd")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-unyt/package.py:21]: Need build dep on setuptools and build/run dep on Python 3.5+

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 9]
```diff
@@ -16,14 +16,20 @@ class PyYt(PythonPackage):
...
     homepage = "https://yt-project.org"
-    url      = "https://github.com/yt-project/yt/archive/yt-3.5.0.tar.gz"
+    url      = "https://github.com/yt-project/yt/archive/yt-4.0.0.tar.gz"
     git      = "https://github.com/yt-project/yt.git"
 
     maintainers = ['qobilidop']
 
-    version("develop", branch="master")
-    version("develop-4.0", branch="yt-4.0")
+    version("develop", branch="main")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]:
```suggestion
    version("main", branch="main")
```

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 10]
```diff
@@ -16,14 +16,20 @@ class PyYt(PythonPackage):
...
+    version("develop", branch="master", deprecated=True)
+    version("develop-4.0", branch="yt-4.0", deprecated=True)
 
+    version('4.0.3', sha256='b0366ccafae1945e4bbf236316968a02968333f2dce558d81c4556fefb2f3dc5')
+    # Prior 4.0.x versions have different dependencies that I don't want to suport.
+    # version('4.0.2', sha256='217a694e9288cba762f3870cae1d575303214bc5c5c79dd4e9f8a3b9b144ed0a')
+    # version('4.0.1', sha256='da9f9b03a3fe521396006a0fdc0e10c1eba09113d9ac0a40455a299694230104')
+    # version('4.0.0', sha256='c5b4e5101e223284bf0c5ab37934f6d5b5dece0a1bb0db152a629f4229c31105')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: You can prob just delete these lines

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 11]
```diff
@@ -48,6 +54,7 @@ class PyYt(PythonPackage):
     depends_on("py-cython@0.24:", type=('build', 'run'))
     depends_on("py-h5py@3.1:", type=('build', 'run'), when="+h5py")
     depends_on("py-ipython@1.0:", type=('build', 'run'))
+    depends_on("py-ipython@1.0:", type=('build', 'run'), when="@:4.0.0")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: I think you need to update the previous line, and use `@:3`. Currently all versions still depend on ipython

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 12]
```diff
@@ -57,6 +64,15 @@ class PyYt(PythonPackage):
...
     depends_on("python@2.7.0:2.7,3.5:", type=('build', 'run'))
+    depends_on("py-cmyt@0.2.2:", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-more-itertools@8.4.0:", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-numpy@1.13.3:", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-pyparsing", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-toml", type=('build', 'run'), when="@4.0.0:4.0.2")
+    depends_on("py-tomli@1.2.3:", type=('build', 'run'), when="@4.0.3:")
+    depends_on("py-tomli-w@0.4.0:", type=('build', 'run'), when="@4.0.3:")
+    depends_on("py-tqdm@3.4.0:", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-unyt@2.8.0:", type=('build', 'run'), when="@4.0.0:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Missing build dep on wheel 0.36.2+, need to update cython, python, matplotlib, etc. deps. If the deps have drastically changed you might find it easier to do something like:
```python
with when('@:3'):
    # old deps

with when('@4:'):
    # new deps
```

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 13]
```diff
@@ -71,3 +87,4 @@ def check_install(self):
         # yt = Executable(join_path(prefix.bin, "yt"))
         # yt("--help")
         python(join_path(self.prefix.bin, "yt"), "--help")
+        python("-c", "import yt; print(yt.__version__)")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Import tests are already automatically added, see `lib/spack/spack/build_systems/python.py`.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by charmoniumQ — COMMENTED** [review]
Fixed comments from code review

**Review by charmoniumQ — COMMENTED** [review]
Fix formatting

File: var/spack/repos/builtin/packages/py-cmyt/package.py [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    """Matplotlib colormaps from the yt project !"""
+    homepage = "https://yt-project.org"
+    pypi = "cmyt/cmyt-1.0.4.tar.gz"
+    git = "https://github.com/yt-project/cmyt.git"
+
+    maintainers = ['qobilidop', 'charmoniumq']
+
+    version("main", branch="main")
+
+    version("1.0.4", "c833ed24eadca8c788e59ce27119d26b2e7d4c6af83293e69385412629294b2a")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmyt/package.py]: PyPI checksums are different than GitHub, you'll need to recompute these. You can use `spack checksum py-cmyt` to do this.

File: var/spack/repos/builtin/packages/py-unyt/package.py [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+    """
+    homepage = "https://yt-project.org"
+    pypi = "unyt/unyt-2.8.0.tar.gz"
+    git = "https://github.com/yt-project/unyt.git"
+
+    maintainers = ['qobilidop', 'charmoniumq']
+
+    version("main", branch="main")
+    version("2.8.0", sha256="4fc95704cba4527604f21724d0d9845aea07831a6a2a3d1be9b9d18395e0c345")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-unyt/package.py]: Same here

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 16]
```diff
@@ -22,8 +22,9 @@ class PyYt(PythonPackage):
 
     maintainers = ["qobilidop"]
 
-    version("develop", branch="master")
-    version("develop-4.0", branch="yt-4.0")
+    version("main", branch="main")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Dependencies on main are quite different than the current deps, should probably update them, not sure what release they'll land in though. If you add the latest release, it might have the same deps. This is optional though, your change isn't making the package any worse, it was already incorrect to begin with. This could happen in a separate PR.

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Just need to update the checksums.

To fix the style tests, you can either run `spack style --fix` or @ spackbot fix style (without the space after @)

File: var/spack/repos/builtin/packages/py-tomli/package.py [hunk 17]
```diff
@@ -12,8 +12,8 @@ class PyTomli(PythonPackage):
     Tomli is fully compatible with TOML v1.0.0."""
 
     homepage = "https://github.com/hukkin/tomli"
-    url = "https://files.pythonhosted.org/packages/py3/t/tomli/tomli-1.2.2-py3-none-any.whl"
-    list_url = "https://pypi.org/simple/tomli/"
+    pypi = "https://pypi.org/project/tomli/"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tomli/package.py]: Was this change intentional? I don't think `pypi` supports wheels at the moment, you have to use `url`. Either way, I would save these changes for a separate PR since they aren't related to yt.

**Comment by qobilidop** [comment]
@charmoniumQ Do you want to replace me as the maintainer of the py-yt package? I am no longer actively getting involved with yt. I think you'll be a much better fit for this role with your current PR. So feel free to edit the maintainer config in this PR. Thanks!

File: var/spack/repos/builtin/packages/py-yt/package.py (hunk 16 above)
Comment (charmoniumQ) [reply var/spack/repos/builtin/packages/py-yt/package.py]: I've updated to support 4.1.2 instead.

**Comment by adamjstewart** [comment]
Looks like a bad rebase.

**Comment by charmoniumQ** [comment]
Bad rebase fixed; this PR only wants to merge 12 commits to 4 files.

File: var/spack/repos/builtin/packages/py-tomli/package.py (hunk 17 above)
Comment (charmoniumQ) [reply var/spack/repos/builtin/packages/py-tomli/package.py]: Changed back to `pypi`

File: var/spack/repos/builtin/packages/py-tomli/package.py [hunk 18]
```diff
@@ -12,24 +12,30 @@ class PyTomli(PythonPackage):
...
+    pypi = "tomli/tomli-2.0.1.tar.gz"
+    git = "https://github.com/hukkin/tomli.git"
+
+    maintainers = ["charmoniumq"]
 
     version(
         "2.0.1",
-        sha256="939de3e7a6161af0c887ef91b7d41a53e7c5a1ca976325f429cb46ea9bc30ecc",
+        sha256="de526c12914f0c550d15924c62d72abc48d6fe7364aa87328337a31007fe8a4f",
         expand=False,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tomli/package.py]: You can drop `expand=False` if you're using `.tar.gz` files now. I think the reason we used to use wheels was because `py-flit-core` had a circular dep on tomli, but that's no longer the case, so I'm fine with this change.

File: var/spack/repos/builtin/packages/py-cmyt/package.py [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+
+    maintainers = ["charmoniumq"]
+
+    version("main", branch="main")
+
+    version("1.1.2", sha256="7027514a89331ee5cd672999e34c15feae218c8ed9b127832b6618c6771a869e")
+
+    # https://github.com/yt-project/cmyt/blob/v1.1.2/pyproject.toml#L2
+    depends_on("py-setuptools@40.9:", type="build")
+    depends_on("py-wheel", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmyt/package.py]: Technically wheel is added in the base class so you don't have to add it here

File: var/spack/repos/builtin/packages/py-cmyt/package.py [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+
+    # https://github.com/yt-project/cmyt/blob/v1.1.2/setup.cfg#40
+    depends_on("python@3.8:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmyt/package.py]:
```suggestion
    depends_on("python@3.8:", type=("build", "run"))
```
I'm guessing

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 21]
```diff
@@ -17,14 +17,16 @@ class PyYt(PythonPackage):
...
+    maintainers = ["charmoniumq"]
 
-    version("develop", branch="master")
-    version("develop-4.0", branch="yt-4.0")
+    version("main", branch="main")
+    version("develop", branch="master", deprecated=True)
+    version("develop-4.0", branch="yt-4.0", deprecated=True)
 
+    version("4.1.2", sha256="0ae03288b067721baad14c016f253dc791cd444a1f2dd5d804cf91da622a0c76")
     version("3.6.1", sha256="a1be3ea7e18729d3cd86e9234dc4731bf23200dff3344fa756fe173ea36cc747")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: If you're going to switch to PyPI for all downloads, you'll need to update the checksum for all older versions. Something like:
```console
$ spack checksum py-yt $(spack versions --safe-only py-yt)
```
should do the trick.

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 22]
```diff
@@ -45,19 +47,45 @@ class PyYt(PythonPackage):
...
-    depends_on("py-ipython@1.0:", type=("build", "run"))
-    depends_on("py-ipython@:6", type=("build", "run"), when="^python@:2")
+    # Main dependencies:
+    # See https://github.com/yt-project/yt/blob/yt-4.1.2/setup.cfg#L40
+    depends_on("py-cmyt@0.2.2:", type=("build", "run"), when="@4.1.2:")
+    depends_on("py-ipywidgets@8:", type=("build", "run"), when="@4.1.2")
     depends_on("py-matplotlib@1.5.3:", type=("build", "run"))
     depends_on("py-matplotlib@:3.2.2", type=("build", "run"), when="@:3.6.0")
+    depends_on("py-matplotlib@3.1:", type=("build", "run"), when="@4.1.2:")
+    conflicts("py-matplotlib@3.4.2", when="@4.1.2:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]:
```suggestion
    conflicts("^py-matplotlib@3.4.2", when="@4.1.2:")
```

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 23]
```diff
@@ -45,19 +47,45 @@ class PyYt(PythonPackage):
...
+    depends_on("py-ipython@:6", type=("build", "run"), when="@:3 ^python@:2")
     depends_on("python@2.7.0:2.7,3.5:", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"), when="@4.1.2:")
+
+    # Extras:
+    # See https://github.com/yt-project/yt/blob/yt-4.1.2/setup.cfg#L80
+    depends_on("py-h5py@3.1:3", type=("build", "run"), when="+h5py")
+    depends_on("py-scipy@1.5.0:", type=("build", "run"), when="+scipy")
+    depends_on("rockstar@yt", type=("build", "run"), when="+rockstar")
+    depends_on("py-astropy@4.0.1:6.0.0", type=("build", "run"), when="+astropy")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]:
```suggestion
    depends_on("py-astropy@4.0.1:5", type=("build", "run"), when="+astropy")
```

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 24]
```diff
@@ -45,19 +47,45 @@ class PyYt(PythonPackage):
     variant("scipy", default=True, description="enable scipy support")
     variant("rockstar", default=False, description="enable rockstar support")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py:44]: Since you're the new maintainer of this package... I don't love the way variants are done here. I usually try to stick to the same variants as pip. There's no `pip install yt[astropy]`, there's only `pip install yt[full]`. I would personally make a `+full` variant and default to false to minimize the build. But it's up to you now.

Comment (charmoniumQ) [reply var/spack/repos/builtin/packages/py-yt/package.py:44]: Can a variant be deprecated?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-yt/package.py:44]: Not at the moment

Comment (charmoniumQ) [reply var/spack/repos/builtin/packages/py-yt/package.py:44]: Then I will leave it as is.

File: var/spack/repos/builtin/packages/py-yt/package.py (hunk 21 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-yt/package.py]: Still waiting on these new checksums

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 25]
```diff
@@ -17,46 +17,70 @@ class PyYt(PythonPackage):
...
+    depends_on("py-ipywidgets@8:", type=("build", "run"), when="@4.1.2")
     depends_on("py-matplotlib@1.5.3:", type=("build", "run"))
     depends_on("py-matplotlib@:3.2.2", type=("build", "run"), when="@:3.6.0")
+    depends_on("py-matplotlib@3.1:", type=("build", "run"), when="@4.1.2:")
+    conflicts("^py-matplotlib@3.4.2", when="@4.1.2:")
+    depends_on("py-more-itertools@8.4:", when="@4.1.2:")
     depends_on("py-numpy@1.10.4:", type=("build", "run"))
-    depends_on("py-scipy@1.5.0:", type=("build", "run"), when="+scipy")
+    depends_on("py-numpy@1.14.5:", type=("build", "run"), when="@4.1.2:")
     depends_on("py-setuptools@19.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Should setuptools be removed here, or restricted to older versions?

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 26]
```diff
@@ -17,46 +17,70 @@ class PyYt(PythonPackage):
...
+    depends_on("py-tomli-w@0.4:", type=("build", "run"), when="@4.1.2:")
+    depends_on("py-tqdm@3.4.0:", type=("build", "run"), when="@4.1.2:")
+    depends_on("py-unyt@2.8:2", type=("build", "run"), when="@4.1.2:")
+    depends_on("py-importlib-metadata@1.4:", type=("build", "run"), when="@4.1.2: ^python@:3.7")
+    depends_on("py-tomli@1.2.3:", type=("build", "run"), when="@4.1.2: ^python@:3.10")
+    depends_on("py-typing-extensions@4.2:", type=("build", "run"), when="@4.1.2: ^python@:3.7")
+    # See https://github.com/spack/spack/pull/30418#discussion_r863962805
+    depends_on("py-ipython@1.0:", type=("build", "run"), when="@:3")
     depends_on("python@2.7.0:2.7,3.5:", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"), when="@4.1.2:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: These Python lines aren't necessary but you can keep them if you want.

File: var/spack/repos/builtin/packages/py-yt/package.py (hunk 25 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-yt/package.py]: Actually, it looks like it is used at run-time, so maybe just change the dep from `pyproject.toml` below to be build/run?

Comment (charmoniumQ) [reply var/spack/repos/builtin/packages/py-yt/package.py]: I'm not totally sure what you mean; the newest commit changes `setuptools` to be available at runtime.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-yt/package.py]: You interpreted me correctly.

**Review by adamjstewart — APPROVED** [review]

**Comment by charmoniumQ** [comment]
@spackbot run pipeline

**Review by adamjstewart — APPROVED** [review]
